src/node-lib/curriculum-api-2023/queries/lessonListing/lessonListing.query.ts

```diff
@@ -81,11 +83,20 @@ const lessonListingQuery =
   (sdk: Sdk) => async (args: { programmeSlug: string; unitSlug: string }) => {
     const res = await sdk.lessonListing(args);
 
-    const [unit] = res.unit;
+    const modifiedUnit = applyGenericOverridesAndExceptions<
+      LessonListingQuery["unit"][number]
+    >({
+      journey: "pupil",
+      queryName: "pupilLessonListingQuery",
+      browseData: res.unit,
+    });
 
-    if (!unit) {
+    if (modifiedUnit.length === 0) {
       throw new OakError({ code: "curriculum-api/not-found" });
     }
+
+    const [unit] = modifiedUnit;
+
     const unitLessons = getTransformedLessons(res);
```

Comment:
The overrideds and exceptions will have no effect as you are not using the data in modifiedUnit.  I think all you need to do is pipe the unit that you destructured in getTransformedLessons

Reply:
oops!